docs/dev_guides/settings_validation.md

```diff
@@ -0,0 +1,142 @@
+
+This folder is comprised of three scripts, each of which inspects the settings dictionary at different stages of its journey:
+
+* [valid_types.py](https://github.com/moj-analytical-services/splink/blob/32e66db1c8c0bed54682daf9a6fea8ef4ed79ab4/splink/settings_validation/valid_types.py) - This script includes various miscellaneous checks for comparison levels, blocking rules, and linker objects. These checks are primarily performed within settings.py.
+* [settings_validator.py](https://github.com/moj-analytical-services/splink/blob/32e66db1c8c0bed54682daf9a6fea8ef4ed79ab4/splink/settings_validation/settings_validator.py) - This script includes the core `SettingsValidator` class and contains a series of methods that retrieve information on fields within the user's settings dictionary that contain information on columns to be used in training and prediction. Additionally, it provides supplementary cleaning functions to assist in the removal of quotes, prefixes, and suffixes that may be present in a given column name.
+* [column_lookups.py](https://github.com/moj-analytical-services/splink/blob/32e66db1c8c0bed54682daf9a6fea8ef4ed79ab4/splink/settings_validation/column_lookups.py) - This script contains helper functions that generate a series of log strings outlining invalid columns identified within your settings dictionary. It primarily consists of methods that run validation checks on either raw SQL or input columns and assesses their presence in **all** dataframes supplied by the user.
+
+<hr>
+
+## Extending the Settings Validator
+
+Before adding any code, it's initially import to identify and determine whether the checks you wish to add fall into any of the broad buckets outlined in the scripts above.
+
+If you intend to introduce checks that are distinct from those currently in place, it is advisable to create a new script within `splink/settings_validation`.
+
+The current options you may wish to extend are as follows:
+
+### Logging Errors
+
+Logging multiple errrors sequentially without disrupting the program, is a common issue. To enable the logging of multiple errors in a singular check, or across multiple checks, an [`ErrorLogger`](https://github.com/moj-analytical-services/splink/blob/settings_validation_refactor_and_improved_logging/splink/exceptions.py#L34) class is available for use.
+
+The `ErrorLogger` operates in a similar way to working with a list, allowing you to add additional errors using the `append` method. Once you've logged all of your errors, you can raise them with the `raise_and_log_all_errors` method.
+
+??? note "`ErrorLogger` in practice"
+    ```py
+    from splink.exceptions import ErrorLogger
+
+    # Create an error logger instance
+    e = ErrorLogger()
+
+    # Log your errors
+    e.append(SyntaxError("The syntax is wrong"))
+    e.append(NameError("Invalid name entered"))
+
+    # Raise your errors
+    e.raise_and_log_all_errors()
+    ```
```

Comment:
Does it make sense to show what the output of this is? I know it seems trivial, but just for completeness